pkg/provider/file/file.go

```diff
@@ -110,6 +110,17 @@ func (p *Provider) addWatcher(pool *safe.Pool, directory string, configurationCh
 			case <-ctx.Done():
 				return
 			case evt := <-watcher.Events:
+				if evt.Op == fsnotify.Remove {
+					err = watcher.Remove(evt.Name)
+					if err != nil {
+						log.WithoutContext().WithField(log.ProviderName, providerName).Errorf("Could not remove watcher: %s", err)
+					}
+					err = watcher.Add(directory)
+					if err != nil {
+						log.WithoutContext().WithField(log.ProviderName, providerName).Errorf("Could not re-add watcher: %s", err)
```

Comment:
Do we want to add the directory name to the error message ("Could not re-add watcher for %s: %s"), or does it already show up within err itself?